lib/bolt/applicator.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'open3'
+
+module Bolt
+  Task = Struct.new(:name, :implementations, :input_method)
+
+  class Applicator
+    def initialize(inventory, executor)
+      @inventory = inventory
+      @executor = executor
+    end
+
+    def apply(args, apply_body, _scope)
+      raise(ArgumentError, 'apply requires a TargetSpec') if args.empty?
```

Comment:
This will probably never be hit, the grammar requires at least one argument.

I'd like to narrow this down to two arguments sometime soon. Right now it's open-ended.

Reply:
Do we want to do that in the grammar on bolt?
I can see things like no arguments implicitly runs locally or adding a description argument to align with the other functions.

Reply:
Ok, that could be worth allowing zero arguments, I'll need to modify the grammer.

I'm ok adding more, but I think it would still be a friendly experience if we updated the grammar to reflect the number of options we expect to support.